geomesa-feature/src/main/scala/org/locationtech/geomesa/feature/kryo/SimpleFeatureSerializer.scala

```diff
@@ -0,0 +1,346 @@
+        encodings
+    }
+
+  /**
+   * Finds an encoding function based on the input type
+   *
+   * @param clas
+   * @param metadata
+   * @return
+   */
+  def matchEncode(clas: Class[_], metadata: jMap[AnyRef, AnyRef]): (Output, AnyRef) => Unit = clas match {
+
+    case c if classOf[String].isAssignableFrom(c) =>
+      (out: Output, value: AnyRef) => out.writeString(value.asInstanceOf[String])
+
+    case c if classOf[java.lang.Integer].isAssignableFrom(c) =>
+      (out: Output, value: AnyRef) => if (value == null) {
+        out.writeByte(NULL_BYTE)
+      } else {
+        out.writeByte(NON_NULL_BYTE)
+        out.writeInt(value.asInstanceOf[java.lang.Integer])
+      }
+
+    case c if classOf[java.lang.Long].isAssignableFrom(c) =>
+      (out: Output, value: AnyRef) => if (value == null) {
+        out.writeByte(NULL_BYTE)
+      } else {
+        out.writeByte(NON_NULL_BYTE)
+        out.writeLong(value.asInstanceOf[java.lang.Long])
+      }
+
+    case c if classOf[java.lang.Double].isAssignableFrom(c) =>
+      (out: Output, value: AnyRef) => if (value == null) {
+        out.writeByte(NULL_BYTE)
+      } else {
+        out.writeByte(NON_NULL_BYTE)
+        out.writeDouble(value.asInstanceOf[java.lang.Double])
+      }
+
+    case c if classOf[java.lang.Float].isAssignableFrom(c) =>
+      (out: Output, value: AnyRef) => if (value == null) {
+        out.writeByte(NULL_BYTE)
+      } else {
+        out.writeByte(NON_NULL_BYTE)
+        out.writeFloat(value.asInstanceOf[java.lang.Float])
+      }
+
+    case c if classOf[java.lang.Boolean].isAssignableFrom(c) =>
+      (out: Output, value: AnyRef) => if (value == null) {
+        out.writeByte(NULL_BYTE)
+      } else {
+        out.writeByte(NON_NULL_BYTE)
+        out.writeBoolean(value.asInstanceOf[java.lang.Boolean])
+      }
+
+    case c if classOf[UUID].isAssignableFrom(c) =>
+      (out: Output, value: AnyRef) =>  if (value == null) {
+        out.writeByte(NULL_BYTE)
+      } else {
+        out.writeByte(NON_NULL_BYTE)
+        out.writeLong(value.asInstanceOf[UUID].getMostSignificantBits)
+        out.writeLong(value.asInstanceOf[UUID].getLeastSignificantBits)
+      }
+
+    case c if classOf[Date].isAssignableFrom(c) =>
+      (out: Output, value: AnyRef) => if (value == null) {
+        out.writeByte(NULL_BYTE)
+      } else {
+        out.writeByte(NON_NULL_BYTE)
+        out.writeLong(value.asInstanceOf[Date].getTime)
+      }
+
+    case c if classOf[Geometry].isAssignableFrom(c) =>
+      (out: Output, value: AnyRef) => if (value == null) {
+        out.writeInt(0, true): Unit
+      } else {
+        val bytes = WKBUtils.write(value.asInstanceOf[Geometry])
+        out.writeInt(bytes.length, true)
+        out.write(bytes)
+      }
+
+    case c if classOf[jList[_]].isAssignableFrom(c) =>
+      val subtype = metadata.get("subtype").asInstanceOf[Class[_]]
+      val subEncoding = matchEncode(subtype, null)
```

Comment:
Out of curiosity, do we support List[List[...]] or Map[List[...], ...] or other complex combinations?  Either way, it might be good to document that.